File: lab3-07.04.20/src/math/MathFunction.kt
package math

/**
 * Представляет собой характеристику математической функции.
 * @property func осуществляет расчёт неизвестного параметра x по переданному правилу.
 * @version 1.3
 */
@FunctionalInterface
interface MathFunction {

    /**
     * @param x неизвестные переменные функций.
     */
    fun func(vararg x: Double): Double
}

File: lab3-24.04.20/src/math/MathFunction.scala
package math

/**
 * Представляет собой характеристику математической функции.
 * @author Артемий Кульбако.
 * @version 1.3
 */
trait MathFunction {

  /**
   * Вычисляет значение функции.
   * @param x неизвестные переменные функции.
   */
  def func(x: Double*): Double

  /**
   * Вычисляет точку для отрисовки на графике для функции. По-умолчанию вызывает метод func(varargs x: Double)
   * для функций вида y = x, но необходимо переопределяет для функций с несколькими неизвестными переменными.
   * @param x неизвестные переменные функции.
   */
  def getPlotDot(x: Double*): Double = func(x:_*)
}